# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/VideoConduit.cpp
# Commit: eca9af59900f
# Full Hash: eca9af59900f33eb5774019e5b026a5f1b12bf38
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2020-06-18 04:43:29
# Description:
#   Bug 1639734 - Restore check that sink is registered in AddOrUpdateSink; r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78723
# ==============================================================================

diff -r cc9bea1151d3 -r eca9af59900f media/webrtc/signaling/src/media-conduit/VideoConduit.cpp
--- a/media/webrtc/signaling/src/media-conduit/VideoConduit.cpp	Tue Jun 16 00:17:10 2020 +0000
+++ b/media/webrtc/signaling/src/media-conduit/VideoConduit.cpp	Mon Jun 08 19:14:06 2020 +0000
@@ -1847,7 +1847,9 @@
         NS_NewRunnableFunction("WebrtcVideoConduit::UpdateSink",
                                [this, self = RefPtr<WebrtcVideoConduit>(this),
                                 sink, wants = std::move(wants)]() {
-                                 AddOrUpdateSinkNotLocked(sink, wants);
+                                 if (mRegisteredSinks.Contains(sink)) {
+                                   AddOrUpdateSinkNotLocked(sink, wants);
+                                 }
                                }));
     return;
   }
